src/components/button/Button.js

```diff
@@ -0,0 +1,50 @@
+import PropTypes from "prop-types";
+import tw from "twin.macro";
+import theme from "@/themes/theme";
+import styled from "styled-components";
+
+const StyledButton = styled.button(
+  ({ className, styles: { block, color, border, shape, size } = {} }) => [
+    theme.button.default.base,
+    theme.button.default.color,
+    theme.button.default.hover,
+    theme.button.default.focus,
+    block && tw`w-full h-full`,
+    color && theme.button.color[color],
+    theme.button.border.base,
+    border && theme.button.border[border],
+    shape && theme.button.borderRadius[shape],
+    size && theme.button.size[size],
+    className,
+  ]
+);
+
+const Button = ({ as, children, ...props }) => {
+  return (
+    <StyledButton as={as} {...props}>
+      {children}
+    </StyledButton>
+  );
+};
```

Comment:
Please watch this video:  https://www.loom.com/share/a2f5f191bba94df9a4e8b94864a55dfb

Thanks.

Reply:
Yes, you are right, it seems {...props} works under the hood